bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/internal/base/registry/BaseRegistry.java

```diff
@@ -23,31 +24,37 @@
 
 /**
  * <p>
- * {@linkplain Registry} implementation that delegates actual service owning to
- * the incoming {@linkplain Supplier}
+ * {@linkplain Registry} implementation that is filled and managed at runtime
+ * programmatically.
  * </p>
  * 
- * @param <S> type of {@linkplain Service} to spread
+ * <p>
+ * Not thread safe (yet)
+ * </p>
+ * <p>
+ * Null free zone.
+ * </p>
+ * 
+ * @param <S> type of {@linkplain Service} to keep
  */
 @SuppressWarnings("restriction")
-public final class BaseRegistry<I extends ServiceId, S extends Service<I>> implements Registry<I, S> {
+public abstract class BaseRegistry<I extends ServiceId, S extends Service<I>> implements Registry<I, S> {
 
-	private final Supplier<Collection<S>> services;
+	protected final Map<I, S> services;
```

Comment:
What is the reason to have it protected?

Reply:
`RuntimeRegistry` alters this map (registers and unregisters service)